tests/xbot/common/injection/wpi_factories/TestCommonLibFactory.java

```diff
@@ -0,0 +1,31 @@
+package xbot.common.injection.wpi_factories;
+
+import org.junit.Test;
+
+import edu.wpi.first.wpilibj.I2C;
+import xbot.common.controls.sensors.XJoystick;
+import xbot.common.injection.BaseWPITest;
+
+public class TestCommonLibFactory extends BaseWPITest {
+
+    @Test
+    public void makeOneOfEverything() {
+        CommonLibFactory clf = injector.getInstance(CommonLibFactory.class);
+        
+        clf.createPowerDistributionPanel();
+        XJoystick j = clf.createJoystick(1, 12);
+        clf.createEncoder("asdf", 1, 2, 1);
+        clf.createDigitalInput(1);
+        clf.createAnalogInput(1);
+        clf.createXboxController(1);
+        clf.createSolenoid(1);
+        clf.createDigitalOutput(1);
+        clf.createServo(1);
+        clf.createSpeedController(1);
+        clf.createCANTalon(1);
+        clf.createGyro();
```

Comment:
It'd be interesting to see if we could write a checkstyle rule or similar which enforced that everything showed up hear and in both modules...